### DISCUSSION TITLE
Thread '<unnamed>' panicked at 'assertion failed: new_window as usize * self.window_size < self.size'

### DISCUSSION THREAD

**Discussion by pocy9901** [filecoin-project/lotus#7791] [Build and Sync]
get this error wheng sealing:
thread '<unnamed>' panicked at 'assertion failed: new_window as usize * self.window_size < self.size', /home/filcoin/.cargo/registry/src/m-1ecc6299db9ec823/storage-proofs-porep-10.0.0/src/stacked/vanilla/memory_handling.rs:260:9
stack backtrace:
   0:          0x2150a30 - std::backtrace_rs::backtrace::libunwind::trace::hdcf4f90f85129e83
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/../../backtrace/src/backtrace/libunwind
   1:          0x2150a30 - std::backtrace_rs::backtrace::trace_unsynchronized::h2669e30cb82f6732
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:          0x2150a30 - std::sys_common::backtrace::_print_fmt::hfbda19e17f6db318
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:67:5
   3:          0x2150a30 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1a8751bf59281272
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:46:22
   4:          0x21c0c9f - core::fmt::write::h7aa6cd0067dca82a
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/core/src/fmt/mod.rs:1094:17
   5:          0x21451c5 - std::io::Write::write_fmt::hd7dd3a1df9b6befb
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/io/mod.rs:1580:15
   6:          0x215478b - std::sys_common::backtrace::_print::h551e9ec8a9fa8106
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:49:5
   7:          0x215478b - std::sys_common::backtrace::print::ha4b1c5e95fa040b3
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:36:9
   8:          0x215478b - std::panicking::default_hook::{{closure}}::h0b34c9ab7fb9f857
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:208:50
   9:          0x215426d - std::panicking::default_hook::h3067e8318decd17a
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:225:9
  10:          0x2154e7d - std::panicking::rust_panic_with_hook::h81b8facc50f34daa
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:591:17
  11:          0x2154a47 - std::panicking::begin_panic_handler::{{closure}}::ha376ab85d95a000e
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:495:13
  12:          0x2150eec - std::sys_common::backtrace::__rust_end_short_backtrace::h6795c8afdd1a77e6
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:141:18
  13:          0x21549d9 - rust_begin_unwind
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:493:5
  14:           0x510111 - core::panicking::panic_fmt::hbe99dddd3092ba3c
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/core/src/panicking.rs:92:14
  15:           0x51005d - core::panicking::panic::h3de4db67bd397eb3
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/core/src/panicking.rs:50:5
  16:          0x1f8e4c5 - storage_proofs_porep::stacked::vanilla::memory_handling::IncrementingCursor::increment::h126646fb848c850e
  17:          0x1f850b0 - storage_proofs_porep::stacked::vanilla::create_label::multi::create_label_runner::h090aca36f9914ee4
  18:          0x1f94404 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h0579751fe9341df7
  19:          0x1f898e2 - std::sys_common::backtrace::__rust_begin_short_backtrace::h962705ad3b4aedc9
  20:          0x1f7ff43 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hf1264bdcae671a27
  21:          0x2163947 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h3aa31cb6360b59d9
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/alloc/src/boxed.rs:1546:9
  22:          0x2163947 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h7719d3c7c5841461
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/alloc/src/boxed.rs:1546:9
  23:          0x2163947 - std::sys::unix::thread::Thread::new::thread_start::hfbe13ead469fd0bc
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys/unix/thread.rs:71:17
  24:     0x7fd013431609 - start_thread
  25:     0x7fd012f8d293 - clone
  26:                0x0 - <unknown>
thread '<unnamed>' panicked at 'assertion failed: new_window as usize * self.window_size < self.size', /home/filcoin/.cargo/registry/src/m-1ecc6299db9ec823/storage-proofs-porep-10.0.0/src/stacked/vanilla/memory_handling.rs:260:9
stack backtrace:
   0:          0x2150a30 - std::backtrace_rs::backtrace::libunwind::trace::hdcf4f90f85129e83
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/../../backtrace/src/backtrace/libunwind
   1:          0x2150a30 - std::backtrace_rs::backtrace::trace_unsynchronized::h2669e30cb82f6732
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:          0x2150a30 - std::sys_common::backtrace::_print_fmt::hfbda19e17f6db318
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:67:5
   3:          0x2150a30 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1a8751bf59281272
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:46:22
   4:          0x21c0c9f - core::fmt::write::h7aa6cd0067dca82a
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/core/src/fmt/mod.rs:1094:17
   5:          0x21451c5 - std::io::Write::write_fmt::hd7dd3a1df9b6befb
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/io/mod.rs:1580:15
   6:          0x215478b - std::sys_common::backtrace::_print::h551e9ec8a9fa8106
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:49:5
   7:          0x215478b - std::sys_common::backtrace::print::ha4b1c5e95fa040b3
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:36:9
   8:          0x215478b - std::panicking::default_hook::{{closure}}::h0b34c9ab7fb9f857
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:208:50
   9:          0x215426d - std::panicking::default_hook::h3067e8318decd17a
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:225:9
  10:          0x2154e7d - std::panicking::rust_panic_with_hook::h81b8facc50f34daa
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:591:17
  11:          0x2154a47 - std::panicking::begin_panic_handler::{{closure}}::ha376ab85d95a000e
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:495:13
  12:          0x2150eec - std::sys_common::backtrace::__rust_end_short_backtrace::h6795c8afdd1a77e6
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys_common/backtrace.rs:141:18
  13:          0x21549d9 - rust_begin_unwind
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/panicking.rs:493:5
  14:           0x510111 - core::panicking::panic_fmt::hbe99dddd3092ba3c
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/core/src/panicking.rs:92:14
  15:           0x51005d - core::panicking::panic::h3de4db67bd397eb3
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/core/src/panicking.rs:50:5
  16:          0x1f8e4c5 - storage_proofs_porep::stacked::vanilla::memory_handling::IncrementingCursor::increment::h126646fb848c850e
  17:          0x1f850b0 - storage_proofs_porep::stacked::vanilla::create_label::multi::create_label_runner::h090aca36f9914ee4
  18:          0x1f94404 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h0579751fe9341df7
  19:          0x1f898e2 - std::sys_common::backtrace::__rust_begin_short_backtrace::h962705ad3b4aedc9
  20:          0x1f7ff43 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hf1264bdcae671a27
  21:          0x2163947 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h3aa31cb6360b59d9
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/alloc/src/boxed.rs:1546:9
  22:          0x2163947 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h7719d3c7c5841461
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/alloc/src/boxed.rs:1546:9
  23:          0x2163947 - std::sys::unix::thread::Thread::new::thread_start::hfbe13ead469fd0bc
                               at /rustc/bb491ed23937aef876622e4beb68ae95938b3bf9/library/std/src/sys/unix/thread.rs:71:17
  24:     0x7fd013431609 - start_thread
  25:     0x7fd012f8d293 - clone
  26:                0x0 - <unknown>

**Comment by LexLuthr** [comment]
@pocy9901 Can you please provide details of your miner/worker and configuration? Have you made any changes to the code?
Which version of lotus are you running? Can you please also give details on the which step are you seeing this error?

**Reply by pocy9901** [reply]
I do not change any code.

lotus version 1.13.0+mainnet+git.7a55e8e89
lotus-worker version 1.13.0+mainnet+git.7a55e8e89
lotus-miner version 1.13.0+mainnet+git.7a55e8e89.dirty

Just runing lotus-worker run --addpiece=false --precommit1=true --precommit2=true --commit=true --parallel-fetch-limit=5
and waiting 10mins.


> @pocy9901 Can you please provide details of your miner/worker and configuration? Have you made any changes to the code? Which version of lotus are you running? Can you please also give details on the which step are you seeing this error?

**Reply by LexLuthr** [reply]
Is this a new miner or older one? If this is old then when did you first encounter this error?
if this is a new miner, then you can please update to 1.13.1 for all node/miner/worker. Make sure to update all dependencies as well.

The error is coming from the below function and points to a rust issue.

fn advance_rear_window(&self, new_window: usize) {
        assert!(new_window as usize * self.window_size < self.size);

        let replace_idx = (new_window % 2) as usize;

        let new_buf = Self::map_buf(
            (new_window * self.window_size) as u64,
            self.window_size as usize,
            &self.file,
        )
        .expect("map_buf failed");

        unsafe {
            self.get_mut_bufs()[replace_idx] = new_buf;
        }
    }
}

**Comment by pocy9901** [comment]
An old miner,but for new woker. Not for all workers,just two of them,and installed in this week

**Reply by LexLuthr** [reply]
@pocy9901 Can you try to update the lotus to latest stable release and check if the problem still persists?

**Reply by pocy9901** [reply]
Not work,same error.

**Reply by LexLuthr** [reply]
Can you get me full logs from the worker/miner and details about the sector where you got this error? Is this happening for all new sectors? Can you also tell on which step of sealing the error happens?

**Reply by LexLuthr** [reply]
I am archiving this discussion as there has been no new updates from the author.
Please refer to https://github.com/filecoin-project/lotus/discussions/7228 for the discussion guidelines.
